packages/core/src/routes/swagger.ts

```diff
@@ -1,51 +1,83 @@
 import { toTitle } from '@silverhand/essentials';
-import { IMiddleware } from 'koa-router';
+import Router, { IMiddleware } from 'koa-router';
 import { OpenAPIV3 } from 'openapi-types';
 
 import { isGuardMiddleware, WithGuardConfig } from '@/middleware/koa-guard';
 import { zodTypeToSwagger } from '@/utils/zod';
 
 import { AnonymousRouter } from './types';
 
-export default function swaggerRoutes<T extends AnonymousRouter>(router: T) {
+type HttpMethod = 'get' | 'head' | 'patch' | 'post' | 'delete';
+
+type RouteObject = {
+  path: string;
+  method: HttpMethod;
+  operation: OpenAPIV3.OperationObject;
+};
+
+type MethodsObject = Partial<Record<HttpMethod, OpenAPIV3.OperationObject>>;
+
+type PathsObject = Record<string, MethodsObject>;
+
+const buildOperation = (stack: IMiddleware[], path: string): OpenAPIV3.OperationObject => {
+  const guard = stack.find((function_): function_ is WithGuardConfig<IMiddleware> =>
+    isGuardMiddleware(function_)
+  );
+  const body = guard?.config.body;
+
+  return {
+    tags: [toTitle(path.split('/')[1] ?? 'General')],
+    requestBody: body && {
+      required: true,
+      content: {
+        'application/json': {
+          schema: zodTypeToSwagger(body),
+        },
+      },
+    },
+    responses: {
+      '200': {
+        description: 'OK',
+      },
+    },
+  };
+};
+
+export default function swaggerRoutes<T extends AnonymousRouter, R extends Router<unknown, any>>(
+  router: T,
+  allRouters: R[]
+) {
   router.get('/swagger.json', async (ctx, next) => {
-    const routes = ctx.router.stack.map(({ path, stack, methods }) => {
-      const guard = stack.find((function_): function_ is WithGuardConfig<IMiddleware> =>
-        isGuardMiddleware(function_)
-      );
+    const routes = allRouters.flatMap<RouteObject>((router) =>
+      router.stack.flatMap<RouteObject>(({ path, stack, methods }) =>
+        methods
+          // There is no need to show the HEAD method.
+          .filter((method) => method !== 'HEAD')
+          .map((method) => ({
+            path: `/api${path}`,
+            method: method.toLowerCase() as HttpMethod,
+            operation: buildOperation(stack, path),
+          }))
+      )
+    );
 
-      return { path, methods, guard };
-    });
+    // Group routes by path
+    // eslint-disable-next-line unicorn/prefer-object-from-entries
+    const paths = routes.reduce<PathsObject>(
```

Comment:
Set a map outside and use for .. in ... to loop through the routes? Avoid disabling this lint rule.

Reply:
Refactored. PTAL